### PR TITLE
feat: add older versions compatibility on kube-control-plane and kube-node-common roles

### PR DESCRIPTION
**Pull request by nutellinoit**

### CODE REVIEW

File: docs/COMPATIBILITY_MATRIX.md [hunk 1]
```diff
@@ -11,13 +11,14 @@
...
 
 - :white_check_mark: Compatible
 - :warning: Has issues
 - :x: Incompatible
 
 Notes:
 
-
 - `v1.23.12-rev.1` fixes an issue with yum-versionlock on RHEL systems, Ansible Roles can be used with `1.21.14`, `1.22.13` and `1.23.12` Kubernetes versions.
+- `v1.25.6-rev.1` adds the possibility to install older Kubernetes versions.
```
Comment (ralgozino) [inline docs/COMPATIBILITY_MATRIX.md:23]:
```suggestion
- `v1.25.6-rev.1` adds the possibility to install older Kubernetes versions. ⚠️ Notice that 1.19 is not compatible with `containerd`.
```

File: docs/releases/v1.25.6-rev.1.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+| [etcd](roles/etcd)                             | `3.4.7`                  | `No update`      |
+| [haproxy](roles/haproxy)                       | `2.6`                    | `No update`      |
+| [containerd](roles/containerd)                 | `1.6.8`                  | `No update`      |
+| [kube-node-common](roles/kube-node-common)     | `-`                      | `Updated`        |
+| [kube-control-plane](roles/kube-control-plane) | `-`                      | `Updated`        |
+| [kube-worker](roles/kube-worker)               | `-`                      | `No update`      |
+
+## Update Guide 🦮
+
+There are no changes for existing clusters, only the possibility to install also versions:
```
Comment (ralgozino) [inline docs/releases/v1.25.6-rev.1.md:22]:
```suggestion
There are no changes for existing clusters, only added the possibility to install also the following versions:
```

File: roles/kube-control-plane/defaults/main.yml [hunk 3]
```diff
@@ -41,9 +41,17 @@ upgrade: False
...
+  "1.22.13":
+    kubernetes_image_registry: "registry.k8s.io"
   "1.23.12":
-    kubernetes_image_registry: "k8s.gcr.io"
+    kubernetes_image_registry: "registry.k8s.io"
   "1.24.7":
-    kubernetes_image_registry: "k8s.gcr.io"
+    kubernetes_image_registry: "registry.k8s.io"
   "1.25.6":
     kubernetes_image_registry: "registry.k8s.io"
```
Comment (ralgozino) [inline roles/kube-control-plane/defaults/main.yml:57]: nitpick: missing new line at the EOF

File: roles/kube-node-common/defaults/main.yml [hunk 4]
```diff
@@ -15,6 +15,14 @@ skip_kubelet_upgrade: False
...
 dependencies:
   # To pin dependencies for each Kubernetes version
+  "1.19.16":
+    critools_version: "1.25.0"
+  "1.20.15":
+    critools_version: "1.25.0"
+  "1.21.14":
+    critools_version: "1.25.0"
+  "1.22.13":
+    critools_version: "1.25.0"
```
Comment (ralgozino) [inline roles/kube-node-common/defaults/main.yml:25]: just curious, is critools backwards compatible like that? I could not find anything confirming it in a quick look a their repo

**Review by ralgozino — CHANGES_REQUESTED** [review]

File: roles/kube-node-common/defaults/main.yml (hunk 4 above)
Comment (nutellinoit) [reply roles/kube-node-common/defaults/main.yml:25]: Yes, it should, the compatibility `should` be 1:1 but there is no 1:1 on package managers, and it should not be a problem (the only problem we had was between 1.24 and 1.25, i think)

**Comment by nutellinoit** [comment]
Superseded by #57

**Pull request closed without merging**